src/Onnx/OnnxToFeld.hs

```diff
@@ -227,51 +246,155 @@ mkInitReadFile hf initGroups = L.concat $ start : map initVec initGroups ++ map
         end = L.unlines
                 [ ""
                 , "  fclose(wfile);"
-                , "  return 0;"
+                , "  return w;"
                 , "}"
                 ]
         initVec (ti:tis) = "  w->member" <> field <> ".length = " <> show (length tis + 1) <> ";\n"
                          <> "  w->member" <> field <> ".buffer = malloc(sizeof(" <> t <> ") * "
                                <> "w->member" <> field <> ".length);\n"
             where t = fromString $ renderType $ tiToType ti
                   field = show (tiField ti + 1)
-        initVec [] = error "Impossible"
-        weightRecTC = fromString $ renderType $ mkStructType $ map toFieldType initGroups
-        toFieldType (ti:_) = ("member" ++ show (tiField ti + 1), mkAwLType universal $ tiToType ti)
-        toFieldType []     = error "Impossible"
+        initVec [] = error "onnxToFeld.initVec: Impossible"
+
+-- | Construct a main program
+mkMainFile :: FilePath -> FilePath -> FilePath -> L.ByteString -> [V.ValueInfoProto] -> [V.ValueInfoProto]-> L.ByteString
+mkMainFile bname hf weightFile weightRecTC outputs inputs = start
+  where start = L.unlines $
+                [ "#include \"" <> fromString hf <> "\""
+                , ""
+                , "#include <stdlib.h>"
+                , "#include <stdio.h>"
+                , "#include <inttypes.h>"
+                , ""
+                , checkFun
+                , ""
+                , "typedef " <> weightRecTC
+                , "  weight_rec_t;"
+                , ""
+                , "weight_rec_t* read_constants(const char *wfile_name);"
+                , ""
+                , "int main( int argc, char** argv )"
```

Comment:
Not sure if I missed this the first time around but same style comments for the code output here as the previous parts you adjusted.